app/models/manageiq/providers/autosde/storage_manager.rb

```diff
@@ -114,6 +114,14 @@ def count_resources(component = nil)
     end
   end
 
+  def event_where_clause(assoc = :ems_events, storage_systems = nil)
+    if storage_systems && storage_systems != ['']
```

Comment:
What would cause `['']` to be an incoming value?  If you can prevent this from the calling side and accept an empty array instead, then this line can be simplified to the following since `.present?` handle both `nil` and `[]`

```ruby
   if storage_systems.present?
```